tests/e2e.py

```diff
@@ -0,0 +1,204 @@
+      zone=self.zone
+    )
+    log.info('Boot disk successfully copied: {0:s}.'
+             .format(self.boot_disk_copy.name))
+    self.assertIsInstance(self.boot_disk_copy, gcp.GoogleComputeDisk)
+    self.assertTrue(self.boot_disk_copy.name.startswith('evidence-'))
+    self.assertTrue(self.boot_disk_copy.name.endswith('-copy'))
+
+    # Create and start the analysis VM and attach the boot disk
+    log.info('Starting a new instance: {0:s} for forensics analysis.'
+             .format(self.analysis_vm_name))
+    self.analysis_vm, created = gcp.start_analysis_vm(
+      project=self.project_id,
+      vm_name=self.analysis_vm_name,
+      zone=self.zone,
+      boot_disk_size=10,
+      cpu_cores=4,
+      attach_disk=self.boot_disk_copy
+    )
+    log.info('Instance {0:s} started.'
+             .format(self.analysis_vm_name))
+    self.assertIsInstance(self.analysis_vm, gcp.GoogleComputeInstance)
+    self.assertTrue(created)
+    self.assertEqual(self.analysis_vm.name, 'new-vm-for-analysis')
+
+    if self.disk_to_forensic is not None:
+      # Make a copy of another disk of the instance to analyse
+      log.info('{0:s} disk copy started for instance: {1:s}.'
+               .format(self.disk_to_forensic, self.instance_to_analyse))
+      self.disk_to_forensic_copy = gcp.create_disk_copy(
+        src_proj=self.project_id,
+        dst_proj=self.project_id,
+        instance_name=self.instance_to_analyse,
+        zone=self.zone,
+        disk_name=self.disk_to_forensic
+      )
+      log.info('{0:s} disk successfully copied: {1:s}.'.
+               format(self.disk_to_forensic, self.disk_to_forensic_copy.name))
+      self.assertIsInstance(self.disk_to_forensic_copy, gcp.GoogleComputeDisk)
+      self.assertTrue(self.disk_to_forensic_copy.name.startswith('evidence-'))
+      self.assertTrue(self.disk_to_forensic_copy.name.endswith('-copy'))
+
+      # Use existing forensics VM and attach the other disk
+      log.info('Attaching disk {0:s} to existing instance {1:s}.'
+               .format(self.disk_to_forensic_copy.name, self.analysis_vm_name))
+      self.analysis_vm, created = gcp.start_analysis_vm(
+        project=self.project_id,
+        vm_name=self.analysis_vm_name,
+        zone=self.zone,
+        boot_disk_size=10,
+        cpu_cores=4,
+        attach_disk=self.disk_to_forensic_copy
+      )
+      self.assertIsInstance(self.analysis_vm, gcp.GoogleComputeInstance)
+      self.assertFalse(created)
+      self.assertEqual(self.analysis_vm.name, 'new-vm-for-analysis')
+
+    disks = self.analysis_vm.list_disks()
+    log.warning(disks)
+    if self.disk_to_forensic is not None:
+      self.assertEqual(disks,
+                       ['new-vm-for-analysis',
+                        self.boot_disk_copy.name,
+                        self.disk_to_forensic_copy.name])
+    else:
+      self.assertEqual(disks,
+                       ['new-vm-for-analysis',
+                        self.boot_disk_copy.name])
```

Comment:
How is this ever going to be reached if you check for the opposite condition in L102?